internal/xsql/rows.go

```diff
@@ -44,6 +45,25 @@ func (r *rows) Columns() []string {
 	return cs
 }
 
+// Add `ColumnTypeDatabaseTypeName` to support `Xorm`
+// NOTE: Need to optimize somehow? This might take O(|r.Columns()|^2)
```

Comment:
may be first call of r.result.CurrentResultSet().Columns() need to store column types to internal rows cache?
We get O(|r.Columns()|+1)

no need to add some cache now, it is not a problem. But wrong implementation of cache can have bugs, difficult case - behaviour on iterate to next result set